diff --git a/src/api/api.cc b/src/api/api.cc
index 9d1e53d3bb..5588b99edf 100644
--- a/src/api/api.cc
+++ b/src/api/api.cc
@@ -5842,7 +5842,7 @@ bool TryHandleWebAssemblyTrapPosix(int sig_code, siginfo_t* info,
   // pad and lets the process crash. Therefore, only enable trap handlers if
   // the host and target arch are the same.
 #if (V8_TARGET_ARCH_X64 && !V8_OS_ANDROID) || \
-    (V8_HOST_ARCH_ARM64 && V8_TARGET_ARCH_ARM64 && V8_OS_MACOSX)
+    (V8_HOST_ARCH_ARM64 && V8_TARGET_ARCH_ARM64 && V8_OS_MACOSX && !V8_OS_IOS)
   return i::trap_handler::TryHandleSignal(sig_code, info, context);
 #else
   return false;
diff --git a/src/trap-handler/trap-handler.h b/src/trap-handler/trap-handler.h
index a27ea236e7..ecc1832eec 100644
--- a/src/trap-handler/trap-handler.h
+++ b/src/trap-handler/trap-handler.h
@@ -25,7 +25,7 @@ namespace trap_handler {
 #define V8_TRAP_HANDLER_SUPPORTED true
 #elif V8_TARGET_ARCH_X64 && V8_OS_FREEBSD
 #define V8_TRAP_HANDLER_SUPPORTED true
-#elif V8_HOST_ARCH_ARM64 && V8_TARGET_ARCH_ARM64 && V8_OS_MACOSX
+#elif V8_HOST_ARCH_ARM64 && V8_TARGET_ARCH_ARM64 && V8_OS_MACOSX && !V8_OS_IOS
 #define V8_TRAP_HANDLER_SUPPORTED true
 #else
 #define V8_TRAP_HANDLER_SUPPORTED false
