diff --git a/build/config/android/BUILD.gn b/build/config/android/BUILD.gn
index 6d8bec8..b75fcb4 100644
--- a/build/config/android/BUILD.gn
+++ b/build/config/android/BUILD.gn
@@ -105,6 +105,7 @@ config("runtime_library") {
   if (current_cpu == "arm" || current_cpu == "x86") {
     libs += [ "android_support" ]
   }
+  libs += [ "c++_shared" ]
 
   # arm builds of libc++ starting in NDK r12 depend on unwind.
   if (current_cpu == "arm") {
