diff --git a/build/install-build-deps.sh b/build/install-build-deps.sh
index 7f07bcbf0..657f138af 100755
--- a/build/install-build-deps.sh
+++ b/build/install-build-deps.sh
@@ -570,9 +570,9 @@ fi
 if package_exists libinput-dev; then
     dev_list="${dev_list} libinput-dev"
 fi
-if package_exists snapcraft; then
-    dev_list="${dev_list} snapcraft"
-fi
+# if package_exists snapcraft; then
+#     dev_list="${dev_list} snapcraft"
+# fi
 
 # Cross-toolchain strip is needed for building the sysroots.
 if package_exists binutils-arm-linux-gnueabihf; then
