diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index bc3de06d0..1e96de22c 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1254,7 +1254,7 @@ config("compiler_deterministic") {
     # different build directory like "out/feature_a" and "out/feature_b" if
     # we build same files with same compile flag.
     # Other paths are already given in relative, no need to normalize them.
-    if (is_nacl) {
+    if (is_nacl || (is_ios && use_system_xcode)) {
       # TODO(https://crbug.com/1231236): Use -ffile-compilation-dir= here.
       cflags += [
         "-Xclang",
diff --git a/config/c++/BUILD.gn b/config/c++/BUILD.gn
index e7ed69ec8..cf73e4490 100644
--- a/config/c++/BUILD.gn
+++ b/config/c++/BUILD.gn
@@ -121,7 +121,11 @@ config("runtime_library") {
 
     # Make sure we don't link against the system libstdc++ or libc++.
     if (is_clang) {
-      ldflags += [ "-nostdlib++" ]
+      if (target_environment == "simulator" && current_cpu == "arm64") {
+        // noop
+      } else {
+        ldflags += [ "-nostdlib++" ]
+      }
     } else {
       # Gcc has a built-in abs() definition with default visibility.
       # If it was not disabled, it would conflict with libc++'s abs()
