diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 944a039d0..25baa9bbe 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2045,7 +2045,7 @@ if (is_win) {
       # Specifically tell the linker to perform optimizations.
       # See http://lwn.net/Articles/192624/ .
       # -O2 enables string tail merge optimization in gold and lld.
-      "-Wl,-O2",
+      "-Wl,-Oz",
       "-Wl,--gc-sections",
     ]
   }
