diff --git a/BUILD.gn b/BUILD.gn
index a8aa8d8..0d53984 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -3637,6 +3637,28 @@ if (v8_monolithic) {
   }
 }
 
+#shared_library("v8_shared") {
+v8_loadable_module("v8_shared") {
+  defines = [ "BUILDING_V8_SHARED" ]
+
+  sources = [
+    "src/v8dll-main.cc",
+  ]
+
+  deps = [
+    ":v8_base_without_compiler",
+    ":v8_compiler",
+    ":v8_libbase",
+    ":v8_libplatform",
+    ":v8_maybe_snapshot",
+  ]
+
+  # configs -= [ "//build/config/clang:find_bad_constructs" ]
+  # configs += [ ":internal_config" ]
+  # public_configs = [ ":external_config" ]
+  configs = [ ":internal_config" ]
+}
+
 ###############################################################################
 # Executables
 #
@@ -4489,3 +4511,25 @@ if (!build_with_chromium && v8_use_perfetto) {
     }
   }  # host_toolchain
 }  # if (!build_with_chromium && v8_use_perfetto)
+
+v8_executable("hello_world") {
+  defines = [ "USING_V8_SHARED" ]
+  sources = [
+    "hello-world.cc",
+  ]
+  include_dirs = [
+    "/home/kudo/v8/v8",
+    "/home/kudo/v8/v8/include",
+  ]
+
+  lib_dirs = [
+    "/home/kudo/v8/v8/out.v8",
+  ]
+
+  libs = [
+    "/home/kudo/v8/v8/out.v8/libv8_shared.so",
+  ]
+  deps = [
+  ]
+  configs = [ ":external_config" ]
+}
diff --git a/gni/v8.gni b/gni/v8.gni
index 0a120df..b5adb11 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -226,3 +226,15 @@ template("v8_static_library") {
     configs += invoker.configs
   }
 }
+
+template("v8_loadable_module") {
+  loadable_module(target_name) {
+    forward_variables_from(invoker, "*", [ "configs" ])
+    configs -= v8_remove_configs
+    configs += v8_add_configs
+    configs += invoker.configs
+    ldflags = [
+      "-Wl,--whole-archive",
+    ]
+  }
+}
