diff --git a/BUILD.gn b/BUILD.gn
index fddd525297..d1a2abfe8d 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -262,9 +262,7 @@ config("internal_config") {
     ":v8_header_features",
   ]

-  if (is_component_build) {
     defines += [ "BUILDING_V8_SHARED" ]
-  }
 }

 # This config should be applied to code using the libplatform.
@@ -3685,6 +3683,23 @@ v8_static_library("wee8") {
   }
 }

+v8_loadable_module("libv8") {
+  defines = [ "BUILDING_V8_SHARED" ]
+
+  sources = [
+    "src/v8dll-main.cc",
+  ]
+
+  deps = [
+    ":v8",
+    ":v8_libbase",
+    ":v8_libplatform",
+    ":v8_maybe_snapshot",
+  ]
+
+  configs = [ ":internal_config" ]
+}
+
 ###############################################################################
 # Executables
 #
diff --git a/gni/v8.gni b/gni/v8.gni
index 2a691dfa60..bad875becc 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -226,3 +226,12 @@ template("v8_static_library") {
     configs += invoker.configs
   }
 }
+
+template("v8_loadable_module") {
+  loadable_module(target_name) {
+    forward_variables_from(invoker, "*", [ "configs" ])
+    configs -= v8_remove_configs
+    configs += v8_add_configs
+    configs += invoker.configs
+  }
+}
