diff --git a/BUILD.gn b/BUILD.gn
index fddd525297..d1a2abfe8d 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -3491,6 +3491,23 @@ if (v8_monolithic) {
   }
 }

+v8_loadable_module("libv8") {
+  defines = [ "BUILDING_V8_SHARED" ]
+
+  sources = [
+    "src/v8dll-main.cc",
+  ]
+
+  deps = [
+    ":v8",
+    ":v8_libbase",
+    ":v8_libplatform",
+    ":v8_maybe_snapshot",
+  ]
+
+  configs = [ ":internal_config" ]
+}
+
 ###############################################################################
 # Executables
 #
diff --git a/gni/v8.gni b/gni/v8.gni
index 2a691dfa60..bad875becc 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -207,3 +207,12 @@ template("v8_static_library") {
     configs += invoker.configs
   }
 }
+
+template("v8_loadable_module") {
+  loadable_module(target_name) {
+    forward_variables_from(invoker, "*", [ "configs" ])
+    configs -= v8_remove_configs
+    configs += v8_add_configs
+    configs += invoker.configs
+  }
+}
